libraries/thermocouple/Thermocouple.h

```diff
@@ -97,71 +100,95 @@ namespace Thermocouple {
     }
   }
 
-  // namespace Analog { //TMP36
-  //   // calibration = 25 C at 750mV
-  //
-  //   int DTRDY_PIN = 28;
-  //   int RESET_PIN = 29;
-  //
-  //   // first 7 bits for Arduino Wire
-  //   uint8_t ADC2_ADDR = 0b1001000;
-  //
-  //   float tempOverVoltageScale = 1 / 0.01;
-  //   float voltageOffset = 0.75, tempOffset = 25; // 25 C = 0.450 V
-  //
-  //   long rawRead;
-  //   float voltageRead;
-  //   float tempRead;
-  //
-  //   ADS1219 ads(DTRDY_PIN, ADC2_ADDR, NULL);
-  //
-  //   void init() {
-  //     // ads.begin();
-  //
-  //     ads.setConversionMode(CONTINUOUS);
-  //
-  //     ads.setVoltageReference(REF_EXTERNAL);
-  //
-  //     ads.setGain(GAIN_ONE);
-  //
-  //     ads.setDataRate(90);
-  //   }
-  // 
-  //   void readTemperatureData(float *data) {
-  //     rawRead = ads.readData(2); // thermocouple on AIN1 of ADC2
-  //     // Serial.println(rawRead);
-  //     // Serial.println(ads.readData(2));
-  //     // Serial.println(ads.readData(3));
-  //     voltageRead = (float) rawRead * (5.0 / pow(2,23));
-  //     Serial.println(voltageRead);
-  //     tempRead = ((voltageRead - voltageOffset) * tempOverVoltageScale) + tempOffset;
-  //     data[0] = tempRead;
-  //     data[1] = -1;
-  //   }
-  // }
+  namespace Analog { //TMP36
+
+    // calibration = 25 C at 750mV
+
+    ADS1219 ** _adcs;
+
+    vector<int> * _analogInNumMap;
+
+    int _numSensors;
+
+    float tempOverVoltageScale = 1 / 0.01;
+    float voltageOffset = 0.75, tempOffset = 25; // 25 C = 0.450 V
+
+    long rawRead;
+    float voltageRead;
+    float tempRead;
+
+    void init (int numSensors, vector<int> * analogInMap, ADS1219 ** adcs) {
+      _numSensors = numSensors;
+      _analogInNumMap = analogInMap; // since this variable is coming from the config, I don't think I need to copy it, since it's not going to change/ be deallocated
+      _adcs = _adcs;
+    }
```

Comment:
library is initialized w/ all maps: i2c addresses , pin defs (no pins here), any other data.